io/disk/ltp_fs.py

```diff
@@ -0,0 +1,118 @@
+               archive/master.zip"
+        tarball = self.fetch_asset("ltp-master.zip",
+                                   locations=[url], expire='7d')
+        archive.extract(tarball, self.srcdir)
+        ltp_dir = os.path.join(self.srcdir, "ltp-master")
+        os.chdir(ltp_dir)
+        build.make(ltp_dir, extra_args='autotools')
+        ltpbin_dir = os.path.join(ltp_dir, 'bin')
+        os.mkdir(ltpbin_dir)
+        process.system('./configure --prefix=%s' % ltpbin_dir)
+        build.make(ltp_dir)
+        build.make(ltp_dir, extra_args='install')
+
+    def test(self):
+        for val in self.args:
+            self.log.info("creating ext4 filesystem on disk %s" % self.disk)
+            create_fs = "mkfs.%s %s" % (self.fs, self.disk)
+            self.log.info("%s" % create_fs)
+            ret = process.system(create_fs, shell=True)
+            if ret != 0:
+                self.fail("Created filesystem on %s failed" % self.disk)
+            self.log.info("Mounting disk %s to mount location %s"
+                          % (self.disk, self.mount_point))
+            mounting = "mount %s %s" % (self.disk, self.mount_point)
+            self.log.info("%s" % mounting)
+            ret = process.system(mounting, shell=True)
+            if ret != 0:
+                self.fail("Unable to mount to location %s" % self.mount_point)
+            if self.script == 'runltp':
+                logfile = os.path.join(self.logdir, 'ltp.log')
+                failcmdfile = os.path.join(self.logdir, 'failcmdfile')
+                val += (" -q -p -l %s -C %s -d %s"
+                        % (logfile, failcmdfile, self.mount_point))
+            self.log.info("Args = %s" % self.args)
+            ltpbin_dir = os.path.join(self.srcdir, "ltp-master", 'bin')
+            cmd = os.path.join(ltpbin_dir, self.script) + ' ' + val
+            result = process.run(cmd, ignore_status=True)
+            # Walk the stdout and try detect failed tests from lines
+            # like these:
+            # aio01       5  TPASS  :  Test 5: 10 reads and
+            # writes in  0.000022 sec
+            # vhangup02    1  TFAIL  :  vhangup02.c:88:
+            # vhangup() failed, errno:1
+            # and check for fail_statuses The first part contain test name
+            fail_statuses = ['TFAIL', 'TBROK', 'TWARN']
+            split_lines = (line.split(None, 3)
+                           for line in result.stdout.splitlines())
+            failed_tests = [items[0] for items in split_lines
+                            if len(items) == 4 and items[2] in fail_statuses]
+            if failed_tests:
+                self.fail("LTP tests failed: %s" % ", ".join(failed_tests))
+            elif result.exit_status != 0:
+                self.fail("No test failures detected, but LTP finished with %s"
+                          % (result.exit_status))
+            self.log.info("Removing the filesystem created on %s" % self.disk)
```

Comment:
If I have a tearDown(), then the script will compile and install ltp for every argument that I pass. So consciously avoided that.